file.go

```diff
@@ -120,18 +117,20 @@ func extractFiles(zipFilePath, filesToExtractFromZipPath, localPath string) erro
 func MakeGitHubZipFileRequest(gitHubCommit GitHubCommit, gitHubToken string) (*http.Request, error) {
 	var request *http.Request
 
-	var url string
-
+	// This represents either a commit, branch, or git tag
```

Comment:
Ah, again, the if-statements proliferate throughout the code. Check out my comments on the other fetch PR about creating a `GitHubCommit` interface. It could easily have a `GitRef()` method to handle this.